[fan] Update get_preset_mode() to return StringRef (#5883)
OttoWinter/esphomedocs: content/components/fan/_index.md

    }
```

> - `.get_preset_mode()`  : Retrieve the current preset mode of the fan. Returns `std::string_view`.

Author: ```suggestion
- `.get_preset_mode()`  : Retrieve the current preset mode of the fan. Returns `StringRef`.
```

    }
```

- `get_preset_mode()`: Retrieve the current preset mode of the fan. Returns `StringRef`.

> ```cpp

Author: ```suggestion
```yaml
```

    call.perform();
```

- `set_preset_mode()` / `has_preset_mode()`: Set or check the preset mode of the fan.

> ```cpp

Author: ```suggestion
```yaml
```
